src/value.cpp

```diff
@@ -517,25 +522,17 @@ MemRef::getDimsAndLayoutAndElemTy(
 }
 
 pair<expr, expr> MemRef::load(const vector<expr> &indices) {
-  expr idx = to1DIdxWithLayout(indices);
+  auto [idx, inbounds] = to1DIdxWithLayout(indices);
   auto [expr, success] = m->load(bid, offset + idx);
 
-  // check whether indices are inbound.
-  for (int i = 0; i < indices.size(); i ++)
-    success = success && z3::ult(indices[i], getDim(i));
-
-  return {expr, success.simplify()};
+  return {expr, (success && inbounds).simplify()};
 }
 
```

Comment:
The original inbounds check is replaced with `inbounds` information from to1DIdxWithLayout.